pkg/sources/circleci/circleci.go

```diff
@@ -82,41 +115,49 @@ func (s *Source) Init(_ context.Context, name string, jobId sources.JobID, sourc
 
 // Chunks emits chunks of bytes over a channel.
 func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, _ ...sources.ChunkingTarget) error {
-	projects, err := s.projects(ctx)
+	// TODO: list Artifacts, list checkout-keys, list env-variables
+
+	// list all projects
+	projects, err := s.listAllProjects(ctx)
 	if err != nil {
 		return fmt.Errorf("error getting projects: %w", err)
 	}
 
-	var scanned uint64
+	var countOfProjectsScanned uint64
 	scanErrs := sources.NewScanErrors()
 
-	for _, proj := range projects {
+	for _, project := range projects {
 		s.jobPool.Go(func() error {
-			builds, err := s.buildsForProject(ctx, proj)
+			projectBuilds, err := s.listProjectBuilds(ctx, &project)
 			if err != nil {
-				scanErrs.Add(fmt.Errorf("error getting builds for project %s: %w", proj.RepoName, err))
+				scanErrs.Add(fmt.Errorf("error getting builds for project %s: %w", project.RepoName, err))
 				return nil
```

Comment:
The original code does this, but I think instead of bailing on these errors we should probably just `continue`, otherwise we'll skip things we probably could have successfully scanned. This goes for builds, steps, and actions

Reply:
Everything here depends on the previous step. If we're unable to list a specific build, we won't have its buildNum, which is required to fetch its steps. In that case, we log the error and proceed to the next project.